src/Services/Orders/Distribt.Services.Orders.BusinessLogic/OrdersBusinessLogicDependencyInjection.cs

```diff
@@ -1,17 +1,26 @@
 using Distribt.Services.Orders.BusinessLogic.Data.External;
 using Distribt.Services.Orders.BusinessLogic.Services.External;
+using Distribt.Shared.Databases.MongoDb;
 using Distribt.Shared.Setup.Databases;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using MongoDB.Driver;
 
 namespace Distribt.Services.Orders.BusinessLogic;
 
 public static class OrdersBusinessLogicDependencyInjection
 {
-    public static void AddProductService(this IServiceCollection serviceCollection, IConfiguration configuration)
+    public static async Task AddProductService(this IServiceCollection serviceCollection, IConfiguration configuration)
     {
-        serviceCollection.AddDistribtMongoDbConnectionProvider(configuration, "productStore");
-        serviceCollection.AddScoped<IProductRepository, ProductRepository>();
+        await serviceCollection.AddDistribtMongoDbConnectionProvider(configuration, "productStore");
+        serviceCollection.AddScoped<IProductRepository>( (IServiceProvider serviceProvider) =>
+        {
```

Comment:
so there are a couple of things here that are not correct. 

1 - as the `productRepository` needs mongo, you have to do all this dirty, that will apply to any mongo-related service, which makes it a bad idea.
2 - ideally it should be abstrated away.  (like the `mongoprovider` inside `AddDistribtMongoDbConnectionProvider`)

But the most important part is that the repository must be injected and thats it, if the   way mongo is configured is the issue, the solution is to change the mongo implementation, not to change all the services that use the current implementation. 

that is one of the reasons why the `ProductRepository` uses the provider instead of the mongoobject.